client.go

```diff
@@ -29,13 +37,32 @@ type Client struct {
 	deviceID string // needs to be unique for a user+device combo
 }
 
-func NewClient(baseURL, clientName, clientVersion string) *Client {
+// NewClient creates a jellyfin Client using the url provided.
+func NewClient(urlStr, clientName, clientVersion string) (*Client, error) {
```

Comment:
I think the [functional option](https://golang.cafe/blog/golang-functional-options-pattern.html) pattern could be good here to allow overriding the default timeout in the constructor, and is future-proof for other later options. But I'm good with leaving it for a follow-up PR.

Reply:
I agree the functional option pattern is a great idea! I think a future PR would work best to scope it to that work.

I did introduce an error return here in `NewClient` which would technically be breaking since it now returns an error in addition to the `Client` struct. Are you okay with that? I would assume this function would only be called once so it would be a few line update in supersonic.